### DISCUSSION TITLE
Insights user token

### DISCUSSION THREAD

**Discussion by JSShant** [algolia/instantsearch#5907] [Q&A]
When setting insights to true in instantsearch.
` <InstantSearch
             searchClient={searchClient}
                      indexName={indexName}
                      insights={true}
                    >`
How am i able to pass a custom user token coming from cookies to this like what was achievable using the insights middleware

**Comment by Haroenv** [comment]
You can pass options to `insights`:

```jsx
const App = ({ children }) => (
  <InstantSearch
    searchClient={searchClient}
    indexName={indexName}
    insights={{
      insightsInitParams: {
        userToken: 'mytoken',
      },
    }}
  >
    {children}
  </InstantSearch>
);
```

If the value is available after initialisation, you can use `aa('setUserToken', 'xxx')` as well.

**Reply by JSShant** [reply]
![image](https://github.com/algolia/instantsearch/assets/105369293/8a5b44f6-f1f7-4ecd-9c51-724949b31456)
It doesnt exist within insightInitParams, i actually cant see it anywhere?
![image](https://github.com/algolia/instantsearch/assets/105369293/bad96e3f-3418-4142-9bc5-d2e4fe226b96)

**Reply by Haroenv** [reply]
You seem to be on an older version of InstantSearch, the type has changed (and been fixed) in https://github.com/algolia/instantsearch/pull/5848, thus React InstantSearch 7.0.3

The parameters of the `insights` option are exactly the same as those for the middleware are.
